Show "Manage Automations" dropdown when no policies are present

sgress454 opened the pull request:
For #23243 

# Checklist for submitter

<!-- Note that API documentation changes are now addressed by the product design team. -->

- [X] Changes file added for user-visible changes in `changes/`, `orbit/changes/` or `ee/fleetd-chrome/changes`.
  See [Changes files](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/Committing-Changes.md#changes-files) for more information.

## Details

This PR changes the behavior of the Manage Automations dropdown on the Manage Policies page.  Any user that has permission to manage policies will now always see the dropdown.  If there are no policies added for the selected team (or no policies at all, in the case of "All teams" or users on the free tier), the dropdown is disabled with a tooltip.

## Screenshots

**Free tier:**
<img width="753" alt="image" src="https://github.com/user-attachments/assets/37a3b97a-74b3-4495-ace4-bfece30b3822" />

---

**Premium tier, All Teams:**
<img width="736" alt="image" src="https://github.com/user-attachments/assets/bedd9a5f-e2aa-49da-8943-61bc69af9744" />

---

**Premium tier, team selected:**
<img width="744" alt="image" src="https://github.com/user-attachments/assets/e1c2397b-1d19-46f2-b78d-e7a923f91c8f" />

sgress454 commented:
@noahtalerman I only saw tooltip verbiage for one state, so I added variations for the free tier and for when "All Teams" is selected (see screenshots in description).  Let me know if you want changes to the text.

noahtalerman commented:
> @noahtalerman I only saw tooltip verbiage for one state, so I added variations for the free tier and for when "All Teams" is selected (see screenshots in description). Let me know if you want changes to the text.

Hey @sgress454! Up to @marko-lisica (Product Designer for the [user story](https://github.com/fleetdm/fleet/issues/23243)):

![Screenshot 2025-02-12 at 4 51 58 PM](https://github.com/user-attachments/assets/cac0652f-158a-4894-8cd4-9b88310e4373)

marko-lisica commented:
> > @noahtalerman I only saw tooltip verbiage for one state, so I added variations for the free tier and for when "All Teams" is selected (see screenshots in description). Let me know if you want changes to the text.
> 
> Hey @sgress454! Up to @marko-lisica (Product Designer for the [user story](https://github.com/fleetdm/fleet/issues/23243)):
> 
> ![Screenshot 2025-02-12 at 4 51 58 PM](https://private-user-images.githubusercontent.com/47070608/412627244-cac0652f-158a-4894-8cd4-9b88310e4373.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0NDkzNTUsIm5iZiI6MTczOTQ0OTA1NSwicGF0aCI6Ii80NzA3MDYwOC80MTI2MjcyNDQtY2FjMDY1MmYtMTU4YS00ODk0LThjZDQtOWI4ODMxMGU0MzczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEzVDEyMTczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4MGY3ZTQzZTVkZGEyZTU4ZDdkODdjNzdhNWI4MjEyNmMzMThiNDRjNTliYTk5MDhjNTJjNDU1ZTBlYjYwZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NX4fN5XfzzqEDOxIwh9wYpXjrzmL3hdMP4HTR-lWqJ4)

Great catch @sgress454! I just added these 2 cases to Figma [here](https://www.figma.com/design/7Oeb9nPQF51Fgwiik2aeJt/%2323243-Add-pagination-to-automations-modal-on-Policies-and-disable-button-if-not-available?node-id=5461-3308&t=IRu2N3eDxjwISg1o-11)

<img width="800" alt="Screenshot 2025-02-13 at 13 30 30" src="https://github.com/user-attachments/assets/e18dfd82-5fea-4d37-ac96-bd19c7660ddc" />


I noticed that the tooltips on the screenshots don't have a caret (arrow) pointing to the button and a copy doesn't match what's specified in Figma. We use one without caret only with underlined text.

Tooltip w/o caret:
<img width="390" alt="Screenshot 2025-02-13 at 13 31 50" src="https://github.com/user-attachments/assets/4c375884-75a0-40e2-8966-0b9ccdd81b73" />

Tooltip w/ caret:
<img width="333" alt="Screenshot 2025-02-13 at 13 32 56" src="https://github.com/user-attachments/assets/7bc9dcf2-30c8-4fb1-9587-9346d6830d8f" />

sgress454 commented:
Thanks @marko-lisica!  Updated:

* Added carets to tooltips
* Centered tooltip text
* Updated language for free tier / premium tier with team selected

jacobshandling commented on frontend/pages/policies/ManagePoliciesPage/ManagePoliciesPage.tsx:
```diff
@@ -980,6 +985,52 @@ const ManagePolicyPage = ({
+        />
+      </div>
+    );
+    if (!hasPoliciesToAutomateOrDelete) {
```
Our general pattern for this kind of thing would be more like:

```
      const tipContent =
        isPremiumTier && currentTeamId !== APP_CONTEXT_ALL_TEAMS_ID ? (
          <>
            To manage automations add a policy to this team.
            <br />
            For inherited policies select &ldquo;All teams&rdquo;.
          </>
        ) : (
          "To manage automations add a policy."
        );
```
, though I like this `let`-based version too. If you feel like we should shift in that direction in
general, can you bring it up as a topic at frontend sync? Otherwise let's stick with the
ternary-based approach please.

Also, nice effort towards using more semantic HTML! I will say in my experience in the codebase,
using `<p>` tends to introduce more style issues than it solves. I'd support a concerted app-wide effort
to use more semantic markup, though caution that currently it tends to introduce more problems than
it solves.

sgress454 replied:
It needs to be a real container since it has a style associated with it.  I can change it to `<div>`.  I wasn't really trying to make a policy statement, just force of habit 😄

jacobshandling commented on frontend/pages/policies/ManagePoliciesPage/_styles.scss:
```diff
@@ -55,6 +55,10 @@
     }
+
+    &__tooltip {
+      text-align: center;
```
related to the `<p>`, right?

jacobshandling reviewed (changes requested):
The only requested change is for the "top" positioning, everything else is nice-to-have

sgress454 commented on frontend/pages/policies/ManagePoliciesPage/ManagePoliciesPage.tsx:
```diff
@@ -980,6 +985,52 @@ const ManagePolicyPage = ({
+      }
+
+      automationsDropdown = (
+        <TooltipWrapper underline={false} tipContent={tipContent} showArrow>
```
@jacobshandling If we do position top, it gets cut off by the site nav.  And the [tooltip style sheet](https://github.com/fleetdm/fleet/blob/df8e753e719484f6ae7dafe639c09ef87a5e1213/frontend/styles/var/mixins.scss#L240) currently has:
```
z-index: 99; // not more than the site nav
```
which makes that seem intentional.

jacobshandling replied:
Hmm, I'm not sure why I put that there. Since the Figma has the tooltip up (see [here](https://www.figma.com/design/7Oeb9nPQF51Fgwiik2aeJt/%2323243-Add-pagination-to-automations-modal-on-Policies-and-disable-button-if-not-available?node-id=5331-2654&m=dev)) I'd say it's fine to override, either by locally setting z-index, or you might need to set `TooltipWrapper`'s `fixedPositionStrategy` prop

sgress454 commented:
@jacobshandling Updated this to use "top" positioning and replaced `<p>` with `<div>`!

jacobshandling reviewed (approved):
LGTM
